packages/extensions/src/table/table-utils/move-row.ts

```diff
@@ -0,0 +1,77 @@
+import type { Node } from '@prosekit/pm/model'
+import type { Transaction } from '@prosekit/pm/state'
+import {
+  CellSelection,
+  TableMap,
+} from 'prosemirror-tables'
+
+import {
+  convertArrayOfRowsToTableNode,
+  convertTableNodeToArrayOfRows,
+  moveRowInArrayOfRows,
+} from './convert-row-and-table'
+import { getSelectionRangeInRow } from './get-selection-range-in-row'
+import { findTable } from './query'
+
+export interface MoveRowParams {
+  tr: Transaction
+  origin: number
+  target: number
+  select?: boolean
+  pos?: number
+}
+
+/**
+ * Move a row from index `origin` to index `target`.
+ *
+ * @internal
+ */
+export function moveRow(moveRowParams: MoveRowParams): Transaction {
+  const { tr, origin, target, select = true, pos } = moveRowParams
+  const $pos = pos != null ? tr.doc.resolve(pos) : tr.selection.$from
+  const table = findTable($pos)
+  if (!table) return tr
+
+  const indexesOriginRow = getSelectionRangeInRow(origin)(tr)?.indexes
+  const indexesTargetRow = getSelectionRangeInRow(target)(tr)?.indexes
+
+  if (!indexesOriginRow || !indexesTargetRow) return tr
+
+  if (indexesOriginRow.includes(target)) return tr
+
+  const newTable = moveTableRow(table.node, indexesOriginRow, indexesTargetRow, 0)
+
+  const _tr = tr
+    .setTime(Date.now())
+    .replaceWith(
+      table.pos,
+      table.pos + table.node.nodeSize,
+      newTable,
+    )
```

Comment:
Why do we need a new variable `_tr`? `_tr` and `tr` are the same object if I'm not wrong.